src/generator/AutoRest.Go/Model/CompositeTypeGo.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Model;
+using AutoRest.Core.Utilities;
+using AutoRest.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AutoRest.Go.Model
+{
+    /// <summary>
+    /// Defines a synthesized composite type that wraps a primary type, array, or dictionary method response.
+    /// </summary>
+    public class CompositeTypeGo : CompositeType
+    {
+        private bool _wrapper;
+
+        // True if the type is returned by a method
+        // Refactor -> CodeModelTransformer
+        public bool IsResponseType;
+
+        // Name of the field containing the URL used to retrieve the next result set
+        // (null or empty if the model is not paged).
+        // Refactor -> This is used for too many things... Ideally it should be in the CodeModelTransformer, and later used in the namer 
+        public string NextLink;
+
+        // Refactor -> CodeModelTransformer
+        public bool PreparerNeeded = false;
+
+        public CompositeTypeGo()
+        {
+
```

Comment:
Eeeew. no. :p

---
In reply to: [92900124](https://github.com/Azure/autorest/pull/1684#discussion_r92900124) [](ancestors = 92900124)